File: tabulate-core/src/test/java/io/github/voytech/tabulate/core/builder/fluent/support/OutputStreamTestResultProvider.java
package io.github.voytech.tabulate.core.builder.fluent.support;

import io.github.voytech.tabulate.template.result.ResultProvider;
import org.jetbrains.annotations.NotNull;

import java.io.OutputStream;

public class OutputStreamTestResultProvider implements ResultProvider<TestRenderingContext, OutputStream> {

    @NotNull
    @Override
    public Class<OutputStream> outputClass() {
        return OutputStream.class;
    }

    @Override
    public void setOutput(@NotNull TestRenderingContext renderingContext, OutputStream output) { }

    @Override
    public void flush() { }
}

File: tabulate-core/src/test/kotlin/io/github/voytech/tabulate/support/OutputStreamTestResultProvider.kt
package io.github.voytech.tabulate.support

import io.github.voytech.tabulate.template.result.ResultProvider
import java.io.OutputStream
import java.util.logging.Logger

class OutputStreamTestResultProvider: ResultProvider<TestRenderingContext, OutputStream> {
    override fun outputClass() = OutputStream::class.java
    override fun setOutput(renderingContext: TestRenderingContext, output: OutputStream) {
        logger.info("This is fake implementation of ResultProvider flushing results into OutputStream")
    }

    override fun flush() {
        logger.info("This is fake implementation of ResultProvider flushing results into OutputStream")
    }

    companion object {
        val logger: Logger = Logger.getLogger(OutputStreamTestResultProvider::class.java.name)
    }
}